sympy/tensor/array/ndim_array.py

```diff
@@ -315,8 +315,11 @@ def f(sh, shape_left, i, j):
 
     def __add__(self, other):
         from sympy.tensor.array.arrayop import Flatten
+        from sympy.matrices import Matrix
 
         if not isinstance(other, NDimArray):
+            if isinstance(other, Matrix):
+                return NotImplemented
             raise TypeError(str(other))
```

Comment:
This should always return `NotImplemented`. We shouldn't raise `TypeError` in methods like `__add__` because then it's impossible to define other classes that can interact with this class.

Reply:
I didn't add TypeError , fixed it anyway.